#65 : desc in app.py and fetch in HomeScreen.js

irina-wang opened the pull request:
**app.py**
- Added to the outcomes endpoint in app.py to calculate the total number of outcomes and then used that info in the frontend of HomeScreen.js for the progress bars; should still work if new outcomes are added
- Use desc and "Outcomes__c" to filter related field
- transform json data to python dictionary(easier to process data without hardcoding)
- process data to get the sum for all outcomes in the Trainee pod

**HomeScreen.js**
- Change to using real user data instead of fake user data by passing in the user parameter in app.py and by using token in HomeScreen.js (also changed from axios to fetch).


-----------------------------------------------------------------------------------
Steps to test PR: 
Login in on app using Yichen's account to test having real user data:
> Username: my.waymao@gmail.com
> password: ZijReFQ@AJYz3jfrA3iz

Then the Home Screen should load all the progress bars with the correct number of total outcomes.
> ( Trainee : 3, Associate: 0, Partner: 0)
-----------------------------------------------------------------------------------

Standard Checklist:
- [X] Does the code work? Does it perform its intended function, the logic is correct etc.
- [X] Have you merged the most recent version of staging into your branch?
- [X] Are all merge conflicts resolved? 
- [X] Can any logging or debugging code be removed? (Any debugging statements should be removed)
- [X] Is the `diff` the **minimum amount** needed to achieve this ticket's goal?

jchin01 commented on backend/app.py:
```diff
@@ -53,26 +52,50 @@ def verify():
-	
     return {"verified": bool(0)} # false
 
 @app.route("/calculateProgressBar")
```
Rename the endpoint to something more specific to the Home Screen. This endpoint will only return the total outcomes of the pods. Potentially name it "/calcProgressHomeScreen"

jchin01 commented on backend/app.py:
```diff
@@ -53,26 +52,50 @@ def verify():
+    # Trainee Pod
+    desc = sf.Trainee_POD_Map__c.describe()
+    field_names_and_labels = [(field['name'], field['label']) for field in desc['fields']]
+    filtered_field_names = [field for field in field_names_and_labels if "Outcomes__c" in field[0]]
```
Outcome__c is now changed to Completed__c

jchin01 commented on frontend/components/pod_components/HomeScreen.js:
```diff
@@ -21,58 +23,74 @@ export default class HomeScreen extends React.Component{
-        .then(response => {
-            let data = response.data;
+        getAccessToken().then(accessToken => 
+            fetch(server_add + '/calculateProgressBar', {
```
change to reflect the new endpoint name

titapa-punpun reviewed (changes requested):
See Jackie's comments

titapa-punpun commented on frontend/components/pod_components/HomeScreen.js:
```diff
@@ -1,10 +1,12 @@
 import ProgressBar from '../ProgressBar.js';
+import Axios from 'axios';
 
 const   TRAINEE_TOTAL_OUTCOMES = 7
 const ASSOCIATE_TOTAL_OUTCOMES = 11
```
get this information from the backend instead of hard coding

irina-wang commented:
I changed things mentioned in the comment and updated Abe's design. Currently we don't have data from the partner pod, so I comment out the part in the code -- should work fine when the data is up in the sf.

jchin01 commented on backend/app.py:
```diff
@@ -114,14 +113,118 @@ def verify():
+    # print(Partner_outcome_sum)
+    
+    # TODO:  Partner Pod
+    sum_pod_outcome = {
+        'Trainee_complete': Trainee_outcome_sum,
+        'Trainee_total': Trainee_total_count,
+        'Associate_complete': Associate_outcome_sum,
+        'Associate_total': Associate_total_count,
+        # 'Partner_complete': Partner_outcome_sum,
+        'Partner_total': Partner_total_count
+    }    
```
Nice job cleaning up the data and only passing back to the frontend what is necessary

jchin01 commented on frontend/components/pod_components/HomeScreen.js:
```diff
@@ -6,83 +6,91 @@ import { getAccessToken } from '../../utils/auth.js';
-        .catch(function (error) {
-            console.log(error);
+        .catch(function (error){
+            print("There is no pod data to display. Please contact your manager or More Than Words administrator if you think this is an error.")
```
good error catching

jchin01 commented on frontend/components/pod_components/HomeScreen.js:
```diff
@@ -6,83 +6,91 @@ import { getAccessToken } from '../../utils/auth.js';
-                </Text>
-                <ProgressBar progress={this.state.Trainee_progress} total_outcomes={TRAINEE_TOTAL_OUTCOMES} />
-            </TouchableOpacity>
+        let T_displayBlock = styles.Block;
+        let A_displayBlock = styles.Block;
+        let P_displayBlock = styles.Block;
+        let T_displayBlockText = styles.BlockText;
+        let A_displayBlockText = styles.BlockText;
+        let P_displayBlockText = styles.BlockText;
+        if (this.state.Trainee_progress < this.state.Trainee_total){ // highlight Trainee
+            T_displayBlock = styles.highlightBlock;
+            T_displayBlockText = styles.highlightBlockText;
+        } else if(this.state.Associate_progress < this.state.Associate_total){ // highlight Associate
+            A_displayBlock = styles.highlightBlock;
+            A_displayBlockText = styles.highlightBlockText;
+        } else { // highlight Partner
+            P_displayBlock = styles.highlightBlock;
+            P_displayBlockText = styles.highlightBlockText;
+        }
```
Very interesting way you were able to get the highlighting for the current pod!

jchin01 commented on frontend/components/pod_components/HomeScreen.js:
```diff
@@ -6,83 +6,91 @@ import { getAccessToken } from '../../utils/auth.js';
+        <SafeAreaView style={styles.container}>
+        <TouchableOpacity 
+            style={T_displayBlock} 
+            onPress={() => this.props.navigation.navigate('Trainee Pod', {
+                pod: 'Trainee'
+            })}
```
This is the code that I added in order for the focus area page to be able to navigate to the outcomes screen because you need to pass the current pod as a prop

jchin01 reviewed (approved):
Good work cleaning up the Homescreen, adding styling, and fetching the total and completed outcomes from salesforce. I like how you commented throughout the code, it made it really easy to see what the code was doing so I could check through. Keep this up. Next time be sure to test that all parts of the app still work with your changes, at the very least click through all the pages and make sure they are working as intended without erroring.
